### PR TITLE
fix mysql serverspec to avoid gcache.page by:

### PR DESCRIPTION
**Pull request by Xiaohua-Shen**
- no serverspec scan for gcache related files
- restart mysql

### CODE REVIEW

**Comment by Xiaohua-Shen** [comment]
ok to test

File: roles/percona-server/tasks/serverspec.yml [hunk 1]
```diff
@@ -8,3 +8,5 @@
             dest=/etc/serverspec/spec/localhost/
             mode=0755
   with_fileglob: ../templates/etc/serverspec/*
+  notify:
+    - restart mysql server
```
Comment (nirajdp76) [inline roles/percona-server/tasks/serverspec.yml]: why do we need to notify restart mysql? We are just writing our the checks. It is triggered by sensu client.

**Comment by nirajdp76** [comment]
ok to test

**Comment by nirajdp76** [comment]
"add to whitelist"

**Comment by nirajdp76** [comment]
"ok to test"

**Comment by Xiaohua-Shen** [comment]
1. sensu-client no need to restart, because the serverspec_check.json file not changed.  The serverspec_check.json has command of ```/etc/sensu/plugins/check-serverspec.rb``` and this command will use mysql_spec.rb as config to run.   so next interval checking will use the latest mysql_spec.rb
2. the reason for mysql restart is to make sure existing gcache.pages file are unlocked.  All my test include steps to restart mysql.

I will add another test to see if it works without mysql restart.

**Comment by nirajdp76** [comment]
in that case we should notify mysql restart only when (upgrade | default('False') | bool)

**Comment by nirajdp76** [comment]
"add to whitelist"

**Comment by nirajdp76** [comment]
"ok to test"

**Comment by nirajdp76** [comment]
retest

**Comment by masteinhauser** [comment]
ok to test

**Comment by Xiaohua-Shen** [comment]
ok to test

File: roles/percona-server/tasks/serverspec.yml (hunk 1 above)
Comment (nirajdp76) [inline roles/percona-server/tasks/serverspec.yml]: Let's remove restart mysql from here as this is only needed one time for 3.1.1 update. Probably good to create a separate playbook fix-mysql-gcache.yml

**Comment by Xiaohua-Shen** [comment]
remove serverspec mysql_config check for log-error file and log-warning level, which seems to connect to mysql to collect info and compare.

**Comment by Xiaohua-Shen** [comment]
ok to test

**Comment by masteinhauser** [comment]
I confirmed `mysql_config` does initiate a connection to the mysql server itself.
https://github.com/mizzy/serverspec/blob/377641274f9b9d418040353d7e4a1ed01219db85/lib/serverspec/type/mysql_config.rb#L4

**Pull request merged**